### PR TITLE
[docs] Add recipe for sorting row groups by the number of child rows

### PR DESCRIPTION
**Pull request by cherniavskii**
Use case: https://discord.com/channels/1131323012554174485/1131329358062178324/1169324963740794960

Preview: https://deploy-preview-11164--material-ui-x.netlify.app/x/react-data-grid/recipes-row-grouping/#sorting-row-groups-by-the-number-of-child-rows

### CODE REVIEW

**Comment by cherniavskii** [comment]
Needs https://github.com/mui/mui-x/pull/11792

File: docs/data/data-grid/recipes-row-grouping/RowGroupingSortByChildRows.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+          rowGrouping: {
+            model: ['company', 'cinematicUniverse'],
+          },
+          sorting: {
+            sortModel: [
+              { field: GRID_ROW_GROUPING_SINGLE_GROUPING_FIELD, sort: 'desc' },
+            ],
+          },
+        }}
+        groupingColDef={{
```
Comment (MBilalShafi) [inline docs/data/data-grid/recipes-row-grouping/RowGroupingSortByChildRows.tsx:29]: Should we stabilize the reference to this prop?
1. [`getGroupingColDefs`](https://github.com/mui/mui-x/blob/next/packages/grid/x-data-grid-premium/src/hooks/features/rowGrouping/useGridRowGroupingPreProcessors.ts#L53) might recompute on each render which may yield some rendering side effects.
2. In docs, we [suggest](https://next.mui.com/x/react-data-grid/faq/#how-to-pass-props-to-the-data-grid-for-better-performance) to keep stable references for all non-primitive props, so it'd be better to follow the same in our demos.

**Review by MBilalShafi — APPROVED** [review]
Nice one, I couldn't grab it right now but I remember discussing this use case on some GH issue too 👍

**Review by romgrk — APPROVED** [review]

File: docs/data/data-grid/recipes-row-grouping/RowGroupingSortByChildRows.tsx (hunk 1 above)
Comment (cherniavskii) [reply docs/data/data-grid/recipes-row-grouping/RowGroupingSortByChildRows.tsx]: Done 👍🏻

**Pull request merged**